src/libcommonserver/io/iohelper.cpp

```diff
@@ -478,6 +482,30 @@ bool IoHelper::createDirectory(const SyncPath &path, IoError &ioError) noexcept
     return creationSuccess;
 }
 
+bool IoHelper::deleteDirectory(const SyncPath &path, IoError &ioError) noexcept {
+    std::error_code ec;
+    const bool removalSuccess = std::filesystem::remove_all(path, ec);
```

Comment:
You cast a `[std::uintmax_t](http://en.cppreference.com/w/cpp/types/integer)` into a boolean value. Are you sure you are not getting `removalSuccess`=`true` in case of error?
The overload that takes error_code& argument returns static_cast<[std::uintmax_t](http://en.cppreference.com/w/cpp/types/integer)>(-1) on error.

Reply:
Absolutely! Corrected in next commit.